### ISSUE TITLE
Task - Alerts & Notices: Last Respondent not working [1.10]

### ISSUE THREAD

**Issue by ntozier** [osTicket/osTicket#3647]
In short 
"When I create and assign a new task, the assigned Agent is notified via Email. If I then type in a message, the assigned agent is notified via Email. If the assigned agent responds to my message, I never receive an email alert. I've set the alerts under Settings > Task > Alerts & Notices to notify the Last Respondent under New Activity Alert."

Reported here:
http://osticket.com/forum/discussion/89749/last-respondent-not-notified-when-replying-to-a-task

@JediKev confirmed this bug

**Comment by JediKev** [comment]
So after looking into this deeper, it is technically not a bug. Let’s say I login as Agent 1, create a Task and Assign to Agent 2, as Agent 1 I post an update which alerts Agent 2 (the Assigned Agent), I login as Agent 2, go to the Task, post an update as Agent 2, and now Agent 2 is the last respondent, not Agent 1…so when the system grabs the Last Respondent for the Activity Notice it is Agent 2 which is who posted the update so the system will not send out an alert.

**Code Reference:**
[`postReply()`](https://github.com/osTicket/osTicket/blob/develop-next/include/class.task.php#L978-L1033)
```php
function postReply($vars, &$errors, $alert = true) {
...
    $this->lastrespondent = $response->staff;
...
    $this->onActivity( array(
....
```

**TLDR;**
The `lastrespondent` is set to the responding agent before the `onActivity()` method is called so technically Agent 2 is the Last Respondent, not Agent 1.

**Comment by jfbkbh** [comment]
Thank you, @JediKev, for this info - but if this is the case, the following options for the alerts don't make any sense, do they? The "last respondent" would always be the agent triggering the alert. They don't need the alert, but the "previous last respondent" does.

/scp/settings.php?t=tasks#alerts
![grafik](https://user-images.githubusercontent.com/10819374/79050209-dad2b700-7c28-11ea-9848-2dc29a28f2b1.png)

**Comment by brunomilet** [comment]
I agree with @jfbkbh 

The last respondent will never be alerted, right?

My case:
I post an update, my friend don't receive an email, but i notified him, out of osticket. My friend post an update and (again) nothing about email notification for me.

Can you reopen this issue or reply with an example @JediKev, please ?

Thank you!

**Comment by JediKev** [comment]
@ntozier @brunomilet @jfbkbh

Try this out:
- https://github.com/osTicket/osTicket/pull/5554

Cheers.

**Comment by brunomilet** [comment]
@JediKev, now I get the emails. However when I try to append an answer by replying the email, a new ticket is opened. =/

Like it: https://github.com/osTicket/osTicket/issues/4304

**Comment by JediKev** [comment]
@brunomilet 

As stated in the footer of the alert, you are not supposed to reply to that alert.
<img width="488" alt="Screen Shot 2020-06-29 at 13 31 40" src="https://user-images.githubusercontent.com/11823401/86042551-f15d0300-ba0c-11ea-8b23-f00729e16433.png">

Cheers.
